python/pyspark/sql/dataframe.py

```diff
@@ -1422,6 +1422,52 @@ def colRegex(self, colName: str) -> Column:
         jc = self._jdf.colRegex(colName)
         return Column(jc)
 
+    def convert(self, schema: StructType) -> "DataFrame":
```

Comment:
unfortumately, `as` is a keyword in Python, so just name it `convert` for now...

Reply:
Maybe let's use other names ... for example .. `DataFrame.asType` or `DataFrame.asSchema`